File: backend/lib/vis/uuid_test.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

using Generated.Validation.Utils;


namespace Generated.Validation
{
    public static class Validators
    {
        public static async Task<ValidationResult> ValidateUuid_form(Dictionary<string, object?> data)
        {
            var result = new ValidationResult
            {
                Success = false,
                Errors = new List<ValidationError>(),
                Data = data
            };

            var successData = new Dictionary<string, object?>();
            bool flag = true;
            {
                flag = true;bool flag = true;

if (!data.TryGetValue("session_id", out var rawValue) || rawValue == null)
{
    result.Errors.Add(new ValidationError
    {
        Path = "session_id",
        Message = "Invalid value."
    });
    flag = false;
}
else if (rawValue is not string value)
{
    result.Errors.Add(new ValidationError
    {
        Path = "session_id",
        Message = "Invalid value."
    });
    flag = false;
}
else
{
    bool uuidFlag = true;
    if (!UuidValidator.Validate(value))
    {
        uuidFlag = false;
    }
    else if (UuidValidator.GetVersion(value) != 4)
    {
        uuidFlag = false;
    }

    if (!uuidFlag)
    {
        result.Errors.Add(new ValidationError
        {
            Path = "session_id",
            Message = "Invalid UUID version"
        });
        flag = false;
    }
    value = value.Trim();
    data["session_id"] = value;
}
if (flag)
                {
                    successData["session_id"] = data["session_id"];
                }
            }

            if (result.Errors.Count == 0)
            {
                result = new ValidationResult
                {
                    Success = true,
                    Errors = new List<ValidationError>(),
                    Data = successData
                };
            }

            return result;
        }
    }
}

File: frontend/lib/vis/uuid_test.js

import { validateUuid, getUuidVersion } from "./utils/uuid_validator.js";


export const validateUuid_form = async (data) => {
    let result = {
        success: false,
        errors: [],
        data: data
    };

    const successData = {};
    let flag = true;
    {
        flag = true;flag = true;

if (data.session_id === undefined || data.session_id === null) {
    result.errors.push({
        path: "session_id",
        message: "Invalid value."
    });
    flag = false;
} else if (typeof data.session_id !== "string") {
    result.errors.push({
        path: "session_id",
        message: "Invalid value."
    });
    flag = false;
} else {
    let uuidFlag = true;
    if (!validateUuid(data.session_id)) {
        uuidFlag = false;
    }
    else if (getUuidVersion(data.session_id) !== 4) {
        uuidFlag = false;
    }

    if (!uuidFlag) {
        result.errors.push({
            path: "session_id",
            message: "Invalid UUID version"
        });
        flag = false;
    }
    data.session_id = data.session_id.trim();
}
if (flag) {
            successData.session_id = data.session_id;
        }
    }

    if (result.errors.length === 0) {
        result = {
            success: true,
            errors: [],
            data: successData
        };
    }

    return result;
};